### PR TITLE
Complete assets type

### PR DESCRIPTION
**Pull request by shakhzodniyozov**

### CODE REVIEW

File: src/Application/Features/AssetTypes/Commands/CreateAssetTypeCommandValidator.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using FluentValidation;
+
+namespace MRA.AssetsManagement.Application.Features.AssetTypes.Commands;
+
+public class CreateAssetTypeCommandValidator : AbstractValidator<CreateAssetTypeCommand>
+{
+    public CreateAssetTypeCommandValidator()
+    {
+        RuleFor(x => x.Name).NotNull().NotEmpty().Length(2, 128);
+        RuleFor(x => x.ShortName).NotNull().NotEmpty().Length(2, 32);
```
Comment (SidiqovAbbos) [inline src/Application/Features/AssetTypes/Commands/CreateAssetTypeCommandValidator.cs]: Length of Short name must be between 1-6

File: src/Application/Features/AssetTypes/Commands/UpdateAssetTypeCommandValidator.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using FluentValidation;
+
+namespace MRA.AssetsManagement.Application.Features.AssetTypes.Commands;
+
+public class UpdateAssetTypeCommandValidator : AbstractValidator<UpdateAssetTypeCommand>
+{
+    public UpdateAssetTypeCommandValidator()
+    {
+        RuleFor(x => x.Name).NotNull().NotEmpty().Length(2, 128);
+        RuleFor(x => x.ShortName).NotNull().NotEmpty().Length(2, 32);
```
Comment (SidiqovAbbos) [inline src/Application/Features/AssetTypes/Commands/UpdateAssetTypeCommandValidator.cs]: The same here

File: src/Application/Features/AssetTypes/Commands/UpdateAssetTypeCommandValidator.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        RuleFor(x => x.Id).NotNull().NotEmpty();
```
Comment (SidiqovAbbos) [inline src/Application/Features/AssetTypes/Commands/UpdateAssetTypeCommandValidator.cs]: We don't need to check for Id property

File: src/Application/Mappings/AssetTypeProfile.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+using AutoMapper;
+using MRA.AssetsManagement.Application.Features.AssetTypes.Commands;
+using MRA.AssetsManagement.Domain.Entities;
+
+namespace MRA.AssetsManagement.Application.Mappings;
```
Comment (SidiqovAbbos) [inline src/Application/Mappings/AssetTypeProfile.cs]: Move mapping into Features/AssetTypes

File: src/Infrastructure/Data/Configurations/AssetTypeConfiguration.cs [hunk 5]
```diff
@@ -16,6 +15,21 @@ protected override void RegisterClassMap(BsonClassMap<AssetType> classMap)
     {
         base.RegisterClassMap(classMap);
         classMap.MapMember(x => x.Name).SetElementName("name");
-        
+        classMap.MapMember(x => x.ShortName).SetElementName("shortName");
+        classMap.MapMember(x => x.Icon).SetElementName("icon");
+        classMap.MapMember(x => x.Archived).SetElementName("archived");
+    }
+
+    protected override void CreateIndexIfRequired()
```
Comment (SidiqovAbbos) [inline src/Infrastructure/Data/Configurations/AssetTypeConfiguration.cs]: rename to Configure()

File: src/Web/Server/ConfigureServices.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
+using FluentValidation.AspNetCore;
+
+using MRA.AssetsManagement.Infrastructure.Data;
+
+namespace MRA.AssetsManagement.Web.Server;
+
+public static class ConfigureServices
```
Comment (SidiqovAbbos) [inline src/Web/Server/ConfigureServices.cs]: Delete this file
---------------
services.AddFluentValidationAutoValidation(); should be registered in Application layer

File: src/Web/Server/Controllers/AssetTypesController.cs [hunk 7]
```diff
@@ -13,21 +11,43 @@ namespace MRA.AssetsManagement.Web.Server.Controllers;
...
+    public async Task<ActionResult<IEnumerable<AssetType>>> Get()
     {
-        return await _mediator.Send(new GetAssetTypesQuery());
+        return Ok(await _mediator.Send(new GetAssetTypesQuery()));
     }
-    
+
     [HttpPost]
-    public async Task<IActionResult> Post(AssetType type)
+    public async Task<IActionResult> Post(CreateAssetTypeCommand command)
```
Comment (SidiqovAbbos) [inline src/Web/Server/Controllers/AssetTypesController.cs]: Can we return a new created object?

**Review by SidiqovAbbos — COMMENTED** [review]
Did CR!

File: src/Web/Server/Controllers/AssetTypesController.cs (hunk 7 above)
Comment (shakhzodniyozov) [reply src/Web/Server/Controllers/AssetTypesController.cs]: Same for Update() ?

**Comment by shakhzodniyozov** [comment]
I've made corrections.

**Review by SidiqovAbbos — APPROVED** [review]

**Pull request merged**